src/main/java/uk/gov/hmcts/reform/civil/handler/event/BundleCreationTriggerEventHandler.java

```diff
@@ -1,26 +1,87 @@
 package uk.gov.hmcts.reform.civil.handler.event;
 
-import lombok.RequiredArgsConstructor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
 import uk.gov.hmcts.reform.civil.event.BundleCreationTriggerEvent;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.Bundle;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.IdValue;
+import uk.gov.hmcts.reform.civil.model.bundle.BundleCreateResponse;
+import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
+import uk.gov.hmcts.reform.civil.service.bundle.BundleCreationService;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.CREATE_BUNDLE;
 
 @Slf4j
 @Service
-@RequiredArgsConstructor
+@AllArgsConstructor
 public class BundleCreationTriggerEventHandler {
 
-    /**
-    * This method will send notification to applicant and respondent solicitors.
-    * This method will not throw any exception but log warning if there is any error while
-    * sending notification because these email notifications are not business critical
-    * and provided as a courtesy, as the user can log in and see the new uploads.
-    *
-    * @param event EvidenceUploadNotificationEvent
-    */
+    @Autowired
+    private BundleCreationService bundleCreationService;
+    private final CoreCaseDataService coreCaseDataService;
+    private final CaseDetailsConverter caseDetailsConverter;
+
     @EventListener
-    public void sendBundleCreationTrigger(BundleCreationTriggerEvent event) {
+    public void sendBundleCreationTrigger(BundleCreationTriggerEvent event) throws Exception {
+        log.info("EventId in bundle trigger is :" + event.getCaseId());
+        BundleCreateResponse bundleCreateResponse  = bundleCreationService.createBundle(event);
+        log.info("bundle api response : " + new ObjectMapper().registerModule(new JavaTimeModule()).writeValueAsString(bundleCreateResponse));
+        if (null != bundleCreateResponse && null != bundleCreateResponse.getData() && null != bundleCreateResponse.getData().getCaseBundles()) {
```

Comment:
Try short-circuiting this.
Given that the method looks pretty long, it would help with readability.

Reply:
Added comments/javadoc in method for better understanding and removed loggers which no longer required (Added loggers for testing in demo). (Method length is now 28 lines)

Reply:
Resolving this conversation as we refactored this together afterwards.